src/hotspot/share/jfr/support/jfrThreadLocal.cpp

```diff
@@ -100,6 +99,13 @@ const JfrBlobHandle& JfrThreadLocal::thread_blob() const {
   return _thread;
 }
 
+void JfrThreadLocal::initialize_primordial_thread(JavaThread* jt) {
+  assert(jt != nullptr, "invariant");
+  assert(Thread::is_starting_thread(jt), "invariant");
+  assert(jt->threadObj() == nullptr, "invariant");
+  jt->jfr_thread_local()->_jvm_thread_id = 1;
```

Comment:
Is this supposed to reflect the `java.lang.Thread` thread-id? If so this is changing to 3 with the object monitor virtual thread changes.

Reply:
If the primordial thread ID assignment is changed from 1 to 3, then that will have to be updated, of course.

Reply:
What happens to ID 1 and 2? Other threads that need to be tracked, especially in a similar manner that is done here?

Reply:
OK, they are reserved constants that do not denote a thread but particular states.

That's okay from our perspective.